### PR TITLE
Port rel/1.0.0-preview2 to rel/1.0.0

### PR DESCRIPTION
**Pull request by brthor**
@eerhardt @livarcocc @piotrpMSFT 

Take a look at the last commit here, it's the only interesting one

### CODE REVIEW

File: src/dotnet/commands/dotnet-pack/NuGet/ManifestVersionUtility.cs [hunk 1]
```diff
@@ -15,9 +15,6 @@ internal static class ManifestVersionUtility
         public const int TargetFrameworkSupportForDependencyContentsAndToolsVersion = 4;
         public const int TargetFrameworkSupportForReferencesVersion = 5;
         public const int XdtTransformationVersion = 6;
-        // Note that this version should change from 7 to 8 when the PackageType
-        // schema is merged into here
-        public const int ServiceableVersion = 7;
 
         public static int GetManifestVersion(ManifestMetadata metadata)
         {
```
Comment (eerhardt) [inline src/dotnet/commands/dotnet-pack/NuGet/ManifestVersionUtility.cs]: I think we missed one comment from https://github.com/dotnet/cli/pull/3492/files

```
             // Important: always add newer version checks at the top
```

Comment (eerhardt) [inline src/dotnet/commands/dotnet-pack/NuGet/ManifestVersionUtility.cs]: Can you diff this file between rel/1.0.0 and rel/1.0.0-preview2?  They should be identical.  That will tell us if we missed any more.

**Comment by eerhardt** [comment]
Do we want to bump the `InternalAbstractions` and `DependencyModel` version numbers to 1.1.0 with this merge?

@piotrpMSFT @pakrym

File: src/dotnet/commands/dotnet-pack/NuGet/ManifestVersionUtility.cs (hunk 1 above)
Comment (brthor) [inline src/dotnet/commands/dotnet-pack/NuGet/ManifestVersionUtility.cs]: diffed both you're right missed that comment

Comment (brthor) [inline src/dotnet/commands/dotnet-pack/NuGet/ManifestVersionUtility.cs]: fixed.

**Comment by eerhardt** [comment]
> Do we want to bump the InternalAbstractions and DependencyModel version numbers to 1.1.0 with this merge?

I just got the word from @ericstj and @weshaggard that the strategy we should follow is:

* If the changes to the library contain only bugfixes, increment 3rd part
* if the changes to the library add new features or any API, increment the 2nd part

Since we haven't added any new features or API to these assemblies, let's increment the 3rd part for now. This follows with the versioning story for CoreFX, which follows semantic versioning.

**Comment by brthor** [comment]
@eerhardt sounds good, so `1.0.1-beta-*` ? Or should we match with preview3 prerelease?

**Comment by eerhardt** [comment]
> sounds good, so 1.0.1-beta-* 

That would be my vote.  "b" is pretty high in the alphabet.  We should be able to move to most any name we'd want in the future.

**Comment by ericstj** [comment]
Yep, that's what I'm working on for CoreFx.

**Comment by brthor** [comment]
DTH Test failures...

test OSX x64 Release Build please
test   Windows_NT x86 Debug Build please

**Comment by brthor** [comment]
@troydai DTH tests are timing out
http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/rel_1.0.0/job/release_osx_x64_prtest/2283/testReport/junit/Microsoft.DotNet.ProjectModel.Server.Tests/DthTests/DthCompilation_Initialize_UnresolvedDependency_referenceType___Package___testProjectName___IncompatiblePackageSample___expectedUnresolvedDependency___Microsoft_Web_Administration___expectedUnresolvedType___Package__/

**Comment by troydai** [comment]
oh, no. first incident or common?

**Comment by brthor** [comment]
First time I've seen it in a while. @troydai

**Pull request merged**